prime-router/src/main/kotlin/transport/SftpTransport.kt

```diff
@@ -137,15 +137,20 @@ class SftpTransport : ITransport {
         }
 
         fun ls(sshClient: SSHClient, path: String): List<String> {
+            val lsResults = mutableListOf<String>()
             try {
-                val client = sshClient.newSFTPClient()
-                client.use {
-                    val lsResponse = it.ls(path)
-                    return lsResponse.map { l -> l.toString() }.toList()
+                sshClient.use { sshClient ->
+                    sshClient.newSFTPClient().use {
+                        it.ls(path).map { l -> lsResults.add(l.toString()) }
+                    }
                 }
+            } catch (_: java.util.concurrent.TimeoutException) {
+                // do nothing. some servers just take a long time to disconnect
             } finally {
```

Comment:
This is where the magic happens kids. Just eat the exception and move on.

Reply:
`ls` is only called by our checking code, not by the regular sftp push code.  And sftp to TX was failing, apart the from the `ls` check, true?   That may imply that we need to put the empty catch in other places in the code as well.   Have you been able to test pushing actual files to TX?

I'm wary of empty catches - can we add a logger.warn here?

Reply:
Oh, and _excellent_ debugging work!

Reply:
I can add a warn to the logger. Let me test that out this morning, and good catch. I better check sending a test file as well.